### PR TITLE
#54 - Added Manifest.convert() and it's usage in PullImageManifest

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #54 
Added Manifest conversion feature and it's usage in pull image manifest endpoint

### CODE REVIEW

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 11@ce4a654318ed, thread: PQ-CT2E6TK9B
com.zerocracy.farm.ruled.RdRules[130] java.lang.IllegalStateException: http://datum.zerocracy.com/latest/rules/pm/no-lost-estimates.xsl says: The estimate for job "gh:artipie/maven-adapter#74" exists in estimates.xml, but the job is absent in the WBS among 53 jobs.
```

[0.53.7](https://github.com/zerocracy/farm): CID: [2a9a213d-ed3a-44ed-9022-b308053ef63c](https://www.0crat.com/footprint/CT2E6TK9B/2a9a213d-ed3a-44ed-9022-b308053ef63c), Type: "Add job to WBS"

**Comment by olegmoz** [comment]
@g4s8 please take a look at this issue, no reviewer has been assigned for a long time

File: src/main/java/com/artipie/docker/manifest/JsonManifest.java [hunk 1]
```diff
@@ -53,6 +54,20 @@ public JsonManifest(final Content source) {
         return new Json(this.source).object().thenApply(root -> root.getString("mediaType"));
     }
 
+    @Override
+    public CompletionStage<Manifest> convert(final Collection<String> options) {
```
Comment (g4s8) [inline src/main/java/com/artipie/docker/manifest/JsonManifest.java:58]: @olegmoz do we need to implement conversion from one media type to another? Or this method only validates that manifest media type is supported?

File: src/main/java/com/artipie/docker/http/PullImageManifest.java [hunk 2]
```diff
@@ -116,9 +119,15 @@ public Response response(
             }
             final RepoName name = new RepoName.Valid(matcher.group("name"));
             final ManifestRef ref = new ManifestRef.FromString(matcher.group("reference"));
+            final Collection<String> accepts = StreamSupport.stream(headers.spliterator(), false)
```
[the remark this answers is not in the document]
Comment (olegmoz) [reply src/main/java/com/artipie/docker/http/PullImageManifest.java]: @g4s8 that's much better, thanks!

File: src/main/java/com/artipie/docker/manifest/JsonManifest.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artipie/docker/manifest/JsonManifest.java:58]: @g4s8 for now it is just to ensure that manifest format may be accepted by client. But generally it is expected from Docker Registry to be able to convert from one type to another

**Comment by 0crat** [comment]
Job `gh:artipie/docker-adapter#86` is not assigned, can't get performer

**Pull request merged**